blocks/identity-block/features/signup/default.jsx

```diff
@@ -0,0 +1,153 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from '@arc-fusion/prop-types';
+import { isServerSide } from '@wpmedia/engine-theme-sdk';
+import getProperties from 'fusion:properties';
+import getTranslatedPhrases from 'fusion:intl';
+import FormInputField, { FIELD_TYPES } from '../../components/FormInputField';
+import useIdentity from '../../components/Identity';
+import HeadlinedSubmitForm from '../../components/HeadlinedSubmitForm';
+
+import './styles.scss';
+
+function validatePasswordRegex(
+  pwLowercase,
+  pwMinLength,
+  pwPwNumbers,
+  pwSpecialCharacters,
+  pwUppercase,
+) {
+  return `^(?=.*[a-z]{${pwLowercase},})(?=.*[A-Z]{${pwUppercase},})(?=.*\\d{${pwPwNumbers},})(?=.*[@$!%*?&]{${pwSpecialCharacters},})[A-Za-z\\d@$!%*?&]{${pwMinLength},}$`;
+}
+
+const SignUp = ({ customFields, arcSite }) => {
+  let { redirectURL } = customFields;
+  const { redirectToPreviousPage } = customFields;
+  const { locale } = getProperties(arcSite);
+  const phrases = getTranslatedPhrases(locale);
+
+  const { Identity, isInitialized } = useIdentity();
+
+  const [password, setPassword] = useState();
+  const [email, setEmail] = useState();
+  const [passwordRequirements, setPasswordRequirements] = useState({
+    status: 'initial',
```

Comment:
Should we pull this `'initial'` and `'error'` status into reusable constants?
something like
```
export const STATUS_TYPES = {
  INITIAL: 'initial',
  ERROR: 'error',
}
```

Reply:
The biggest concern I have with this is that it's a useState that appears to be a useReducer.  Below is:
```
          setPasswordRequirements({
            pwLowercase,
            pwMinLength,
            pwPwNumbers,
            pwSpecialCharacters,
            pwUppercase,
            status: 'success',
          });
```
and also:
```
  setPasswordRequirements({ status: 'error' })
```

And because these are useStates, they will wholesale overwrite.  I'm not sure this is a concern because the data involved depends on the status.

But as far as an enum, it might add some safety, but I don't think either of these changes are important at this time.